kanidmd/src/lib/be/idl_arc_sqlite.rs

```diff
@@ -1174,148 +1001,94 @@ impl<'a> IdlArcSqliteWriteTransaction<'a> {
+                self.name_cache.insert_dirty(cache_key, cache_value)
+            });
+            Ok(())
         })
     }
 
     // ! TRACING INTEGRATED
-    pub fn write_name2uuid_rem(
-        &mut self,
-        audit: &mut AuditScope,
-        rem: BTreeSet<String>,
-    ) -> Result<(), OperationError> {
+    pub fn write_name2uuid_rem(&mut self, rem: BTreeSet<String>) -> Result<(), OperationError> {
         spanned!("be::idl_arc_sqlite::write_name2uuid_rem", {
-            lperf_trace_segment!(audit, "be::idl_arc_sqlite::write_name2uuid_add", || {
-                // self.db.write_name2uuid_rem(audit, &rem).and_then(|_| {
-                rem.into_iter().for_each(|k| {
-                    // why not just a for loop here...
-                    let cache_key = NameCacheKey::Name2Uuid(k);
-                    self.name_cache.remove_dirty(cache_key)
-                });
-                Ok(())
-                // })
-            })
+            // self.db.write_name2uuid_rem(audit, &rem).and_then(|_| {
+            rem.into_iter().for_each(|k| {
+                // why not just a for loop here...
+                let cache_key = NameCacheKey::Name2Uuid(k);
+                self.name_cache.remove_dirty(cache_key)
+            });
+            Ok(())
+            // })
         })
     }
 
     // ! TRACING INTEGRATED
-    pub fn create_uuid2spn(&self, audit: &mut AuditScope) -> Result<(), OperationError> {
-        self.db.create_uuid2spn(audit)
+    pub fn create_uuid2spn(&self) -> Result<(), OperationError> {
+        self.db.create_uuid2spn()
     }
 
     // ! TRACING INTEGRATED
-    pub fn write_uuid2spn(
-        &mut self,
-        audit: &mut AuditScope,
-        uuid: &Uuid,
-        k: Option<Value>,
-    ) -> Result<(), OperationError> {
+    pub fn write_uuid2spn(&mut self, uuid: &Uuid, k: Option<Value>) -> Result<(), OperationError> {
         spanned!("be::idl_arc_sqlite::write_uuid2spn", {
-            lperf_trace_segment!(audit, "be::idl_arc_sqlite::write_uuid2spn", || {
-                /*
-                self.db
-                    .write_uuid2spn(audit, uuid, k.as_ref())
-                    .and_then(|_| {
-                */
-                let cache_key = NameCacheKey::Uuid2Spn(*uuid);
-                match k {
-                    Some(v) => self
-                        .name_cache
-                        .insert_dirty(cache_key, NameCacheValue::S(Box::new(v))),
-                    None => self.name_cache.remove_dirty(cache_key),
-                }
-                Ok(())
-                /*
-                    })
-                */
-            })
+            let cache_key = NameCacheKey::Uuid2Spn(*uuid);
+            match k {
+                Some(v) => self
+                    .name_cache
+                    .insert_dirty(cache_key, NameCacheValue::S(Box::new(v))),
+                None => self.name_cache.remove_dirty(cache_key),
+            }
```

Comment:
idk if you can tell but I hate `match`ing on `Result`'s and `Option`'s
```suggestion
            if let Some(v) = k {
                self
                    .name_cache
                    .insert_dirty(cache_key, NameCacheValue::S(Box::new(v))),
            } else {
                self.name_cache.remove_dirty(cache_key),
            }
```

Reply:
I know you hate it, but for now I want to keep this more on tracing/audit, not logic changes.